include/deal.II/matrix_free/matrix_free.h

```diff
@@ -2292,7 +2292,10 @@ inline unsigned int
 MatrixFree<dim, Number, VectorizedArrayType>::n_components() const
 {
   AssertDimension(dof_handlers.size(), dof_info.size());
-  return dof_handlers.size();
+  unsigned int comps = 0;
+  for (unsigned int i = 0; i < dof_handlers.size(); i++)
+    comps += dof_handlers[i]->get_fe().n_components();
+  return comps;
```

Comment:
The failing tests are related to these line of codes. @NiklasWik Why did you change these? I know `n_components()` is not a really good name here; it should be `n_dof_handler()`.

Reply:
Ah! I had problems with `FEEvaluationBase.n_components` and read in the documentation that the value comes from matrix_free, found that function by just using ctrl+F, but then realized that the value is set in matrix_free.templates.h and not by this function. Didn't change this back because I figured that all previous cases had `dof_handlers.size() == n_components` anyway. I'll change it back!